ui/raidboss/emulator/data/CombatantJobSearch.ts

```diff
@@ -0,0 +1,133 @@
+export default class CombatantJobSearch {
+  static getJob(abilityID: number): string | undefined {
+    for (const Job in CombatantJobSearch.abilities) {
+      if (CombatantJobSearch.abilities[Job]?.includes(abilityID))
+        return Job;
+    }
+  }
+
+  static readonly abilityMatchRegex = /[a-fA-F0-9]{1,4}/i;
+
+  static readonly abilities: Record<string, number[]> = {
```

Comment:
```suggestion
  static readonly abilities: { [job: string]: number[] } = {
```

I don't think `Record<string, etc>` should get used.  It's really just an index type, and not a `Record`, even if it technically works.

Reply:
I thought in a previous PR you had indicated that `Record` should be used for maps with static pre-defined keys? Or maybe I misunderstood and you meant keys that map to types like `Lang`?

Technically the type could also instead be auto-inferred I think, like this (though I'm not at my desktop to verify right now):
```typescript
static readonly abilities = {
...
};
```

Reply:
I think quis meant if the key type is a built-in basic type, we prefer the indexing approach; if the key type is a user-defined union type, we prefer to use `Record`?

Reply:
Sorry, here's my long-winded reasoning.

For context (sorry if this is repeating what you already know), my understanding is that `Record<Job, number[]>` means something very different to TypeScript than `Record<string, number[]>`.  More than just the type of the key, a `Record` on a union type (or some other type with finite values) also requires that every key is present in the `Record` with a valid value and that there are no keys that are not part of that union type.  I think there's a reason that the TypeScript page itself only lists a `Record` with a union type key as an example.

(The nice part here is that it gives you an error if anybody ever adds an additional `Job` and doesn't update a `Record<Job, etc>`. Arguably, your code here could be `Record<Job, number[]>` but you'd need to `as keyof typeof abilities` like you did in the other patch and also add crafting jobs unless you partitioned `Job` out into `BattleJob` or somesuch.  I'm not sure this is better?  Maybe it is?)

At any rate, my understanding is that `Record<Job, number[]>` is a shorter way to write `{ [key in Job]: number[] }`, My understanding also is that `Record<string, number[]>` is equivalent to all of the following:
1) `{ [key: string]: number[] }`
2) `{ [key: string]?: number[] }`
3) `{ [key in string]: number[] }`
4) `{ [key in string]?: number[] }`

I am not a TypeScript expert, but personally it feels like if you want to say "this is an object with string keys" then (1) is the "idiomatic" way to write that, and unlike `Record` it also gives you a chance to name the key.

To sum up, my feelings here are mostly:
* naming the key is nice
* `Record` behaves specially with union types, and subjectively feels like it should only be used for union types
* if there's 5 ways to do something, let's pick one idiomatic one and do that everywhere